src/reader/_storage/_sqlite_utils.py

```diff
@@ -119,6 +120,10 @@ def wrap_exceptions(
         for fragment, op_exc_type in op_exc_types.items():
             if fragment.lower() in e_msg:
                 raise op_exc_type(str(e)) from None
+
+        if 'readonly' in e_msg:
+            raise exc_type("operation on readonly database") from e
+
```

Comment:
Not needed, the catchall `raise exc_type(message) from e` below handles it automatically.